mdal/frmts/mdal_gdal.cpp

```diff
@@ -564,36 +564,76 @@ std::unique_ptr<MDAL::Mesh> MDAL::DriverGdal::load( const std::string &fileName,
 
     // some formats like NETCFD has data stored in subdatasets
     std::vector<std::string> subdatasets = parseDatasetNames( mFileName );
-
-    // First parse ALL datasets/bands to gather vector quantities
-    // if case they are splitted in different subdatasets
+    // Parse all the dataset
+    // For consistency, search the first dataset with a projection and store it first to use it as reference for the mesh
+    // store other dataset in a std::vector
+    std::vector<std::shared_ptr<MDAL::GdalDataset>> datasets;
+    bool firstProjFound = false;
     for ( auto iter = subdatasets.begin(); iter != subdatasets.end(); ++iter )
     {
       std::string gdal_dataset_name = *iter;
       // Parse dataset parameters and projection
       std::shared_ptr<MDAL::GdalDataset> cfGDALDataset = std::make_shared<MDAL::GdalDataset>();
       cfGDALDataset->init( gdal_dataset_name );
 
-      if ( !mMesh )
+      if ( !firstProjFound && !cfGDALDataset->mProj.empty() )
       {
-        // If it is first dataset, create mesh from it
+        firstProjFound = true;
         gdal_datasets.push_back( cfGDALDataset );
+      }
+      else
+      {
+        datasets.push_back( cfGDALDataset );
+      }
+    }
+
```

Comment:
here I would just add 
```
if !firstProjFound
    take first one
```

and then the rest of the function can be the same for both cases?

Reply:
or last commit?